src/screens/AddNewItemScreen/index.tsx

```diff
@@ -105,8 +167,11 @@ export const AddNewItemScreen: FC = () => {
         onConfirm={onConfirmDate}
         onCancel={onCancelDate}
       />
-      <TouchableOpacity disabled={!text} style={buttonStyle} onPress={onPress}>
-        <Text style={style.text}>{'Add new task'}</Text>
+      <TouchableOpacity
+        disabled={!hasUnsavedChanges}
```

Comment:
your button should be disabled also when request is pending.

how you can see, have to handle a lot of component if you don't use overlay loader